go/types/set_editor.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2017 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package types
+
+import (
+	"sort"
+
+	"github.com/attic-labs/noms/go/d"
+)
+
+// SetEditor allows for efficient editing of Set-typed prolly trees. Edits
+// are buffered to memory and can be applied via Build(), which returns a new
+// Set. Prior to Build(), Get() & Has() will return the value that the resulting
+// Set would return if it were built immediately prior to the respective call.
+// Note: The implementation biases performance towards a usage which applies
+// edits in key-order.
+type SetEditor struct {
+	s          Set
+	edits      setEditSlice // edits may contain duplicate values, in which case, the last edit of a given key is used
+	normalized bool
+}
+
+func NewSetEditor(s Set) *SetEditor {
+	return &SetEditor{s, setEditSlice{}, true}
+}
+
+func (se *SetEditor) Kind() NomsKind {
+	return SetKind
+}
+
+func (se *SetEditor) build(vrw ValueReadWriter) Value {
+	return se.Build(vrw)
+}
+
+func (se *SetEditor) Build(vrw ValueReadWriter) Set {
```

Comment:
What do you think about calling this `Set()` (and `Map()` in the case of `MapEditor`). It is more parallel with the `Value()` method which does the same thing.